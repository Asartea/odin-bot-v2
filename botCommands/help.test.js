const commands = require('./help');

describe('!help', () => {
  it.each([
    ['!help'],
    [' !help'],
    ['!help @odin-bot'],
    ['!help me please'],
    ['!help @user'],
    [' !help @user'],
  ])("'%s' triggers the callback", (string) => {
    expect(commands.help.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['!hmadpofdsnja'],
    ['!halp'],
    ['help'],
    ['!hlpe'],
    ['!he lp'],
    ['! help'],
    ['@odin-bot ! help'],
    ['!he&l^p'],
    ['!h*el&p^'],
    ['!^help'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(commands.help.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['send !help'],
    ['need some !help ?'],
    ['hey @odin-bot, teach me !help'],
    ['!@odin-bot ^ !me !help !test$*'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(commands.help.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['!helpsgrealkdmsfalnd'],
    ['someone!help'],
    ['!help!xx'],
    ['!help*'],
    ['!help...'],
    ['^_!help...'],
  ])(
    "'%s' - command should be its own word!group - no leading or trailing characters",
    (string) => {
      expect(commands.help.regex.test(string)).toBeFalsy();
    },
  );
});

describe('!help snapshot', () => {
  it('should return the correct output', () => {
    expect(commands.help.cb({})).toMatchSnapshot();
  });
});
