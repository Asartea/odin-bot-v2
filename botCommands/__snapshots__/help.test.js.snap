// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/help snapshot should return the correct output 1`] = `
"
    **By posting in this chatroom you agree to our code of conduct:** <https://github.com/TheOdinProject/theodinproject/blob/master/doc/code_of_conduct.md>
  
  Give points to someone who has been helpful by mentioning their name and adding ++ : \`@username ++\` or by giving them a star : \`@username :star:\`
  
  View the points leaderboard with \`/leaderboard\`
  Modify it with \`n=\` and \`start=\` i.e. \`/leaderboard n=25 start=30\`
  
  Type \`/help\` to view this message again
  
  Motivate your fellow odinites with \`/motivate\` and mention them
  
  I'm open source!  Hack me HERE: <https://github.com/TheOdinProject/odin-bot-v2>"
`;
